File: w01/d05/Ann/lab/shelter.rb
# This file is a class definition for Shelter. 
#This shelter has a name. 

require_relative 'animals'

class Shelter
  def initialize( name )
    @name = name
    @animals = {}
    @clients = {}
  end 

  def to_s
    "Welcome to #{@name}!"
  end 

  #This adds an Animal object in to the Shelter object, into a hash named @animals 
  def add_animals( new_animal )
      @animals[ new_animal.name ] = new_animal
  end 

  def return_animal  

  end 

  #This prints the animals in the shelter. 
  def list_animals
    puts "The animals in the shelter are: #{@animals}."  
  end 

  def list_clients
    puts "The clients at the shelter are: #{@clients}."  
  end  

end 

happitrails = Shelter.new( "Happitails!" )
puts happitrails.to_s


happitrails.add_animals( Animal.new( "GOLDIE!", "goldfish" ) )
happitrails.add_animals( Animal.new( "Stew", "sparrow" ) )

happitrails.list_animals

File: w06/d02/Bradley/calc/js/shelter.js
function Shelter(name) {
  this.name = name;
  this.animals = [];
}

Shelter.prototype.adopt = function(animal) {
  this.animals.splice(this.animals.indexOf(animal), 1);
}

Shelter.prototype.toString = function() {
  console.log(this.name + " has " + this.animals.length + " animals up for adoption.");
}

function Animal(name, type) {
  this.name = name;
  this.type = type;
}

Animal.prototype.speak = function() {
  "Hi! I am a " + this.type + " named " + this.name + ".";
}

Shelter.prototype.hasAnimal = function(animal) {
  if (this.animals.indexOf(animal) != -1) {
    false;
  }
  else {
    true;
  }
}